docs: update TRG 7-04 to unbind DEPENDENCIES file from the main branch (#1095)
eclipse-tractusx/eclipse-tractusx.github.io: docs/release/trg-7/trg-7-01.md

title: TRG 7.01 - Legal Documentation
---

| Status | Created     | Post-History                                    |
|--------|-------------|-------------------------------------------------|
| Active | 22-Nov-2024 | Add alternative way to handle DEPENDENCIES file |
| Active | 25-Apr-2024 | Updates for CC-BY-4.0 license                   |
| Active | 24-Aug-2023 | Updated SECURITY.md file                        |
| Active | 20-Jul-2023 | References to TRG 7.07, 7.08 updated            |
| Active | 13-Apr-2023 | Moved from OSS Development                      |

## Why

> Eclipse Tractus-X is an open source project hosted by the Eclipse Foundation licensed under the Apache License 2.0
> ([Apache-2.0](https://spdx.org/licenses/Apache-2.0)). For non-code the default license is the Creative Commons Attribution 4.0 International ([CC-BY-4.0](https://spdx.org/licenses/CC-BY-4.0.html)).

Reviewer: ```suggestion
Eclipse Tractus-X is an open source project hosted by the Eclipse Foundation licensed under the Apache License 2.0 ([Apache-2.0](https://spdx.org/licenses/Apache-2.0)). For non-code the default license is the Creative Commons Attribution 4.0 International ([CC-BY-4.0](https://spdx.org/licenses/CC-BY-4.0.html)).
```

The legal obligations of the content must be observed in all forms of which the content is available.

> This page contains information about legal documentation requirements in your repositories. The source of truth is always
> the [Eclipse Foundation Project Handbook](https://www.eclipse.org/projects/handbook/#legaldoc).

Reviewer: ```suggestion
This page contains information about legal documentation requirements in your repositories. The source of truth is always the [Eclipse Foundation Project Handbook](https://www.eclipse.org/projects/handbook/#legaldoc).
```

- LICENSE
- LICENSE_non-code
- NOTICE.md
- SECURITY.md
- CONTRIBUTING.md
- CODE_OF_CONDUCT.md

> While the following can be omitted if appropriate actions are taken:

Reviewer: ```suggestion
The following file **must** be present on root level for every released version but can be omitted in the main branch if appropriate actions are taken:
```

- Add both licenses to the "Declared Project Licenses" sections, see [example](https://github.com/eclipse-tractusx/sig-infra/blob/main/NOTICE.md)
- Add the link to your repository
- Add the link(s) to your SBOM, e.g. the DEPENDENCIES file (one or more)
- Add information for third party content checks, if not covered by the Dash Tool (e.g. IP checks for icons, fonts, ...)

[Further information](trg-7-04.md#checking-other-content-fonts-images-) and see the [Handbook#legaldoc-notice](https://www.eclipse.org/projects/handbook/#legaldoc-notice).

### DEPENDENCIES FILE

:::info

> Third-party dependencies need to be checked regularly to reflect your code changes. The DEPENDENCIES file must be updated
> accordingly. This is recommended for every contribution (e.g. PR).

Reviewer: ```suggestion
Third-party dependencies need to be checked regularly to reflect your code changes. The DEPENDENCIES file must be updated accordingly. This is recommended for every contribution (e.g. PR).
```

Author: same

:::

- Create it with the [Eclipse Dash License Tool](https://www.eclipse.org/projects/handbook/#ip-license-tool)

> If different technologies / package managers (e.g. npm and maven) are used you are free to have several dependency files.

Reviewer: ```suggestion
If different technologies / package managers (e.g. npm and maven) are used you are free to have several dependencies files.
```

Use the naming convention `DEPENDENCIES_XYZ`, e.g. `DEPENDENCIES_FRONTEND` and `DEPENDENCIES_BACKEND`.

> These files can be kept either checked in the git repository or published to a static location (e.g. GitHub Pages) and
> linked in the [NOTICE file](#notice-file) .
> It is advisable to run the check after every commit or during the nightly, surely it is mandatory to run it before the
> release, ensuring that no `rejected` or `restricted` dependencies are being part of delivered artifacts.

Reviewer: ```suggestion
These files can be kept either checked in the git repository or published to a static location (e.g. GitHub Pages) and linked in the [NOTICE file](#notice-file) during development against the main branch.
It is advisable to run the check after every commit or during the nightly, surely it is mandatory to run it before the release, ensuring that no `rejected` or `restricted` dependencies are being part of delivered artefacts.
```

Author: same

Reviewer: ```suggestion
These files can be kept either checked in the git repository or published to a static location (e.g. GitHub Pages) and
linked in the [NOTICE file](#notice-file).
It is advisable to run the check after every commit or during the nightly, surely it is mandatory to run it before the
release, ensuring that no `rejected` or `restricted` dependencies are being part of delivered artefacts.
```

Author: In the Java world (that's the base of most of the products here) "artifact" is used ([as maven states in their doc](https://maven.apache.org/repositories/artifacts.html)). I would prefer to keep it as it is

Reviewer: sure, could please fix the other thing?

```suggestion
These files can be kept either checked in the git repository or published to a static location (e.g. GitHub Pages) and
linked in the [NOTICE file](#notice-file).
It is advisable to run the check after every commit or during the nightly, surely it is mandatory to run it before the
release, ensuring that no `rejected` or `restricted` dependencies are being part of delivered artifacts.
```

Author: sure, that little whitespace felt through the cracks, generally it would be easier to tackle single line comments unless they are related to a whole block.

done, btw :)
